src/main/java/com/csc530/smtlib/SimpleSolver.java

```diff
@@ -0,0 +1,512 @@
+import org.smtlib.IPos;
+import org.smtlib.IResponse;
+import org.smtlib.IResponse.IAssertionsResponse;
+import org.smtlib.IResponse.IAssignmentResponse;
+import org.smtlib.IResponse.IAttributeList;
+import org.smtlib.IResponse.IProofResponse;
+import org.smtlib.IResponse.IUnsatCoreResponse;
+import org.smtlib.IResponse.IValueResponse;
+import org.smtlib.ISolver;
+import org.smtlib.ISort.IAbbreviation;
+import org.smtlib.ISort.IApplication;
+import org.smtlib.ISort.IFamily;
+import org.smtlib.ISort.IFcnSort;
+import org.smtlib.ISort.IParameter;
+import org.smtlib.ITheory;
+import org.smtlib.IVisitor;
+import org.smtlib.IVisitor.VisitorException;
+import org.smtlib.SMT;
+import org.smtlib.SMT.Configuration;
+
+import com.csc530.sat.DecisionDiagram;
+import com.csc530.sat.DecisionDiagramNode;
+import com.csc530.sat.SATVariable;
+
+class SimpleSolver implements ISolver {
+
+    private final Set<ISymbol> booleanVariables;
+    private final Set<IExpr> assertions;
+    private final SMT.Configuration smtConfig;
+
+    public SimpleSolver(SMT.Configuration smtConfig) {
+        this.smtConfig = smtConfig;
+        booleanVariables = new HashSet<>();
+        assertions = new HashSet<>();
+    }
+
+    @Override
+    public Configuration smt() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public IResponse checkSatStatus() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public IResponse start() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public IResponse exit() {
+        // reset the solver for next use
+        booleanVariables.clear();
+        assertions.clear();
+        return smtConfig.responseFactory.success();
+    }
+
+    @Override
+    public IResponse set_logic(String logicName, IPos pos) {
+        if ("QF_UF".equals(logicName)) {
+            return smtConfig.responseFactory.success();
+        } else {
+            throw new UnsupportedOperationException("Not yet implemented");
+        }
+    }
+
+    @Override
+    public IResponse push(int number) {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public IResponse pop(int number) {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public IResponse assertExpr(IExpr expr) {
+        assertions.add(expr);
+        return smtConfig.responseFactory.success();
+    }
+
+    @Override
+    public IResponse check_sat() {
+        Map<ISymbol, SATVariable> symbolMap = new HashMap<>();
+        for (ISymbol sym : booleanVariables) {
+            symbolMap.put(sym, new SATVariable(sym.value()));
```

Comment:
oh hey look it here it makes no promise of such thing:
https://github.com/smtlib/jSMTLIB/blob/master/SMT/src/org/smtlib/IExpr.java#L112

this will break